iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1645,6 +1646,17 @@ private void updateStatus(IotHubConnectionStatus newConnectionStatus, IotHubConn
             {
                 log.debug("Updating transport status to new status {} with reason {}", newConnectionStatus, reason);
             }
+            else if (this.getDefaultConfig() != null
+                    && !this.getDefaultConfig().isLoggingRoutineDisconnectsAsErrors()
+                    && newConnectionStatus == IotHubConnectionStatus.DISCONNECTED_RETRYING
+                    && reason == IotHubConnectionStatusChangeReason.EXPIRED_SAS_TOKEN
+                    && (protocol == IotHubClientProtocol.MQTT || protocol == IotHubClientProtocol.MQTT_WS))
+            {
+                // This is a special case where the user has opted out of treating the routine
+                // MQTT/MQTT_WS SAS token disconnects as an error for logging purposes. As such,
+                // log at the debug level instead of warn or error level.
+                log.debug("Updating transport status to new status {} with reason {}", newConnectionStatus, reason);
```

Comment:
Maybe it is better to log as info, client must know that a disconnect happens. `isLoggingRoutineDisconnectsAsErrors` is about not log as error.

Reply:
I can make this an info level log, sure